node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py

```diff
@@ -215,3 +191,51 @@ def test_node_declaration_scr_as_other_roles(
             '44f8c702b3e3ca931b5bb6e534781a135d5c17c4ff03886a80f32643dbd8fe0d',
         'signer': '1c8e5f54a15b63a9f3d540ce505fd0799575ffeaac62ce625c917e6d915ea8bb'
     }
+
+
+@pytest.mark.parametrize(
+    'update_with', (
+        {},
+        {
+            'fee': 5
+        },
+        {
+            'fee': 1
+        },
+        {
+            'fee': 0
+        },
+        {
+            'addresses': ['http://not-existing-address.com:5555/']
+        },
+        {
+            'addresses': ['http://not-existing-address.com:5555/', 'http://not-existing-address.com:7777/']
+        },
+        {
+            'fee': 5,
+            'addresses': ['http://not-existing-address.com:5555/']
+        },
+    )
+)
+@pytest.mark.usefixtures('base_blockchain', 'as_primary_validator')
+def test_update_node(update_with, api_client, primary_validator_node, primary_validator_key_pair):
+    blockchain_facade = BlockchainFacade.get_instance()
+    node = blockchain_facade.get_node_by_identifier(primary_validator_node.identifier)
+    assert node.fee == 4
+    assert node.addresses == [
+        'http://not-existing-primary-validator-address-674898923.com:8555/',
+    ]
+
+    updated_node = deep_update(node.dict(), update_with)
+    message = NodeDeclarationSignedChangeRequestMessage(
+        node=updated_node,
+        account_lock=primary_validator_node.identifier,
+    )
+    signed_change_request = SignedChangeRequest.create_from_signed_change_request_message(
+        message=message,
+        signing_key=primary_validator_key_pair.private,
+    )
+    payload = signed_change_request.json()
+    response = api_client.post('/api/signed-change-requests/', payload, content_type='application/json')
+    assert response.status_code == 201
+    assert response.json()['message']['node'] == updated_node
```

Comment:
@olegtropinin this is not enough. You need to use blockchain facade methods to get node and assert for its values

Reply:
Yes, I added blockchain facade method as you said.